geth/transactions/txqueue_manager.go

```diff
@@ -269,19 +275,17 @@ func (m *Manager) DiscardTransactions(ids []common.QueuedTxID) map[common.Queued
 // It accepts one param which is a slice with a map of transaction params.
 func (m *Manager) SendTransactionRPCHandler(ctx context.Context, args ...interface{}) (interface{}, error) {
 	log.Info("SendTransactionRPCHandler called")
-
 	// TODO(adam): it's a hack to parse arguments as common.RPCCall can do that.
 	// We should refactor parsing these params to a separate struct.
 	rpcCall := common.RPCCall{Params: args}
 	tx := common.CreateTransaction(ctx, rpcCall.ToSendTxArgs())
-
 	if err := m.QueueTransaction(tx); err != nil {
 		return nil, err
 	}
-
-	if err := m.WaitForTransaction(tx); err != nil {
-		return nil, err
+	rst := m.WaitForTransaction(tx)
+	if rst.Error != nil {
+		return nil, rst.Error
 	}
-
-	return tx.Hash.Hex(), nil
+	// handle empty hash
```

Comment:
Is it a TODO?

Reply:
i think i forgot to remove, hash cant be empty if err is nil now